substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixUtils.java

```diff
@@ -259,4 +260,24 @@ public static int readBytes(int fd, CCharPointer buffer, int bufferLen, int read
         }
         return readBytes;
     }
+
+    /**
+     * Emulates the deprecated {@code signal} function via its replacement {@code sigaction},
+     * assuming BSD semantics (like glibc does, for example).
+     *
+     * Use this or {@code sigaction} directly instead of calling {@code signal} or {@code sigset}:
+     * they are not portable and when running in HotSpot, signal chaining (libjsig) prints warnings.
+     */
+    public static Signal.SignalDispatcher installSignalHandler(int signum, Signal.SignalDispatcher handler) {
+        Signal.sigaction old = StackValue.get(Signal.sigaction.class);
+        Signal.sigaction act = StackValue.get(Signal.sigaction.class);
+        Signal.sigemptyset(act.sa_mask());
+        Signal.sigaddset(act.sa_mask(), signum);
+        act.sa_flags(Signal.SA_RESTART());
```

Comment:
Quite confusing that the accessors in `sigaction` do not have a `get` and `set` prefix. Not your fault, but you still could do some renamings if you want.